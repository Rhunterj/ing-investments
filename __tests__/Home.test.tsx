
import { render, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils';
import Home from '../pages/index'
import { StockItemsMock } from './__mocks__/StockItemsMock';
import { tasksHandlerException } from './__mocks__/handlers';
import { server } from './__mocks__/server';

describe('Home', () => {
  it('renders a heading', () => {
    act(() => {
      render(<Home data={StockItemsMock}/>)
    })

    const heading = screen.getByRole('heading', {
      name: 'AEX',
    })

    expect(heading).toBeInTheDocument()
  });

  it('Failed api reqeust', async () => {
    server.use(tasksHandlerException);

    render(<Home />);

    expect(await screen.findByText('There was an issue collecting the data')).toBeInTheDocument();
  });

  it('render Home unchanged', () => {
    const { container } = render(<Home data={StockItemsMock} />);
    expect(container).toMatchSnapshot();
  });
})