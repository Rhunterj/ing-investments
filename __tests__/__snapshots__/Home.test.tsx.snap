// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render Home unchanged 1`] = `
<div>
  <div
    class="sc-43a7f46e-0 eJvKtc"
  >
    <main
      class="sc-43a7f46e-1 eqbKnS"
    >
      <h1>
        AEX
      </h1>
      <h2>
        Alle instrumenten in de AEX
      </h2>
      <table
        class="sc-821e15d8-0 Eedrd"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Course
            </th>
            <th>
              Difference
            </th>
            <th>
              Difference in %
            </th>
            <th>
              Time
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="sc-104b807a-0 ejZdDV"
          >
            <td
              class="sc-104b807a-1 ucnII"
            >
              <a
                href="/details/Adyen?name=Adyen"
              >
                Adyen
              </a>
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              1622.365
               EUR
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              +26.16
               EUR
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              +1.64
              %
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              10:38
            </td>
          </tr>
          <tr
            class="sc-104b807a-0 gzYzbh"
          >
            <td
              class="sc-104b807a-1 ucnII"
            >
              <a
                href="/details/Aegon?name=Aegon"
              >
                Aegon
              </a>
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              4.043
               EUR
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              -0.04
               EUR
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              -1.13
              %
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              10:38
            </td>
          </tr>
          <tr
            class="sc-104b807a-0 ejZdDV"
          >
            <td
              class="sc-104b807a-1 ucnII"
            >
              <a
                href="/details/Ahold-Delhaize?name=Ahold+Delhaize"
              >
                Ahold Delhaize
              </a>
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              26.25
               EUR
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              +0.13
               EUR
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              +0.48
              %
            </td>
            <td
              class="sc-104b807a-1 ucnII"
            >
              10:38
            </td>
          </tr>
        </tbody>
      </table>
    </main>
  </div>
</div>
`;
